src/app/content/utils/urlUtils.ts

```diff
@@ -90,3 +99,11 @@ export const toRelativeUrl = (from: string, to: string) => {
   return '../'.repeat(Math.max(0, parsedFrom.length - commonParts.length - 1))
     + parsedTo.slice(commonParts.length).join('/');
 };
+
+export const findPathForParams = (params: object, paths: string[]) => {
```

Comment:
move this, and the guard, into `navigation/utils` and `navigation/guards`

this file is for content urls, but these two functions apply to any routing